app/routes/extension/ConfirmPassword/components/ConfirmPasswordForm/components/Layout.js

```diff
@@ -15,16 +15,16 @@ class Layout extends Component {
 
     return (
       <React.Fragment>
-        <span data-validation={dataValidation}>
+        <div data-validation={dataValidation}>
           <input
             type='password'
             placeholder='Confirm password'
             value={confirmPassword}
             onChange={updateConfirmPassword}
           />
-        </span>
-        <p className={confirmPassword === '' ? null : matchStyle}>
-          Password doesn't match
+        </div>
```

Comment:
Do you use in more places this approach? If you only validate data here, I think... it is kind of ok, if not, I suggest to use a proper library.

Also, I see it complex, so styles are applied when something is wrong, but you show text if is ```! ready``` so that means the same paragraph is used for two things.

Reply:
The text is grey at the begining. When the filed confirmPassword is filled, the text becomes red. Once the expression ready is passed, the text disappears. This is how it was defined by the design team.

Reply:
Why not...
```
{ differentPasswords &&
    <p className={ filling ? redStyle : greyStyle }>
        'Password does not match'
    </p>
}
```